src/Serilog/Serilog.csproj

```diff
@@ -35,19 +36,11 @@
 
   <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' " />
 
-  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.1' ">
-    <DefineConstants>$(DefineConstants);FEATURE_DEFAULT_INTERFACE;FEATURE_SPAN;FEATURE_ITUPLE;FEATURE_ASYNCDISPOSABLE;FEATURE_DICTIONARYTRYADD</DefineConstants>
-  </PropertyGroup>
-
-  <PropertyGroup Condition=" '$(TargetFramework)' == 'net5.0' ">
-    <DefineConstants>$(DefineConstants);FEATURE_DEFAULT_INTERFACE;FEATURE_SPAN;FEATURE_ITUPLE;FEATURE_ASYNCDISPOSABLE;FEATURE_WRITE_STRINGBUILDER;FEATURE_TOHEXSTRING;FEATURE_DICTIONARYTRYADD</DefineConstants>
-  </PropertyGroup>
-
   <PropertyGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
     <DefineConstants>$(DefineConstants);FEATURE_DEFAULT_INTERFACE;FEATURE_SPAN;FEATURE_ITUPLE;FEATURE_DATE_AND_TIME_ONLY;FEATURE_ASYNCDISPOSABLE;FEATURE_WRITE_STRINGBUILDER;FEATURE_TOHEXSTRING;FEATURE_DICTIONARYTRYADD</DefineConstants>
   </PropertyGroup>
 
-  <PropertyGroup Condition=" '$(TargetFramework)' == 'net7.0' ">
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'net8.0' ">
```

Comment:
this flagset is identical to net6.0, which means there is no reason for it to exists save for:
- potentially different packing stuff that might make the DLL smaller (but by how much?)
- potential perf benefit because the compiler can perhaps generate slightly different code targeting new intrinsics or something else unspecific

If it was me, I wouldnt bother actually adding net8.0 here without having a reason, and "it can't hurt and the DLL seems to be 16K smaller" is a bit weak for me

But it's not me, so this is just me attempting to stir debate (and/or some might claim, troll.... :trollface: )

Reply:
>  the DLL seems to be 16K smaller

On that subject, and refing #1999 - Is there actually anything in here that needs pollyfilling in .NET 6.0 anyway?
(A quick tinker with the current dev branch suggests that it builds with the PolySharp NuGet package only being referenced in the .NET FW and Standard build targets)

Reply:
I think the benefit here is just having a simple policy that will work consistently across the majority of the Serilog projects and discourage projects from drifting into odd/arbitrary TFM lists.

We don't use anything .NET 8 specific in this repo (yet) but there's enough new surface are in 7 and 8 that we'll likely consume some of it at some point.

Not a strongly-held opinion, but also keen just to move on - we're hitting diminishing returns on more time investment on this I think :-)

Reply:
If adding 8.0 preemptively makes someone happy, fine by me - the main thing is that there comes a time when it gets trimmed.

But if I can (continue to be) a very pedantic evil's advocate, I'm not sure it's that great an example to be seeding things with - if other lbs take 6+8, you know that people will feel the instinct to add 10.0 to follow the pattern (which is where we started on this) - and at that point, net6.0 will be in the baseline set, and net8.0 and net10.0 will be in LTS.

Only in late 2026 do we get to net8.0 being removed, why which time the it looks like you just add LTSes to the list.

But to be fair, the policy wording is also clear, and deviations can be picked up in code review.

Reply:
It's tricky to balance the need for a clear/simple policy with the unpredictability of what we might want from future .NETs... I don't think it'll be a big burden to see a handful of LTSes in there - but we can reevaluate if it gets out of hand :-)

I find that if I'm not targeting the latest TFM by default while editing code, it's too easy to miss the existence of new APIs via autocomplete etc., so some amount of wanting 8 there is a quality-of-life thing.

Reply:
That's a good point I hadn't considered - the only bit that seems unresolved to me is whether `net6.0` targeting is going to be trimmed in the `net10.0` timeframe.

My general policy atm is to target `net6.0` only and be conservative in adding new TFM-specific builds until it hurts. But then my libs are more I/O bound in nature than Serilog, so it's kinda apples and oranges. Eventually removing `net6.0` from the list makes life a tiny bit more interesting than I'd like it to be. (Having said that, I'm writing libs that should prb=obably target otel, not serilog in the long term)

Can I be so rude as to page you regarding this @andrewlock ?

TL;DR the proposed policy is that on rollign the major version of Serilog going forward, that the TMFs list gets trimmed to `netstandard2.0` + the currently supported LTSes and (optionally) the latest STS if there's something relevant in there that someone has a specific use for in realising a feature (or significant perf win)

Reply:
Just in case you missed it @bartelink, I merged your suggestion to add `net6.0` to the always-included list alongside `netstandard2.0`.

Give it a few years and it's hard to predict what will be optimal, but I think we're on the same page RE `net6.0` being the "new `netstandard`" for the foreseeable future 👍

Reply:
Doh! Yes, my bad; I looked at it wrong, and was interpreting as if you'd opted not to have `net6.0` on the permanent list (without any specific reason).
And you're def not wrong with caution about throwing around forward looking statements!